lib/auth/auth.ts

```diff
@@ -61,7 +61,7 @@ function extractParams(
     log: Logger,
     awsService: string,
     data: { [key: string]: string }
-) {
+): { err: null | Error | ArsenalError, params?: AuthV2RequestParams | AuthV4RequestParams | AuthInfo } {
```

Comment:
can we really return an `Error` ?

Reply:
Through the `checkFunctions`, yes: the `validateCredentials` function in the header+query v4 check can return such error. V2 isn't returning `Error`s indeed.

Reply:
`validateCredentials` is typed to return an `Error`, but implementation looks like it is always returning an `ArsenalError`

Reply:
I updated the code (and all other related comments).